applications/GeoMechanicsApplication/custom_constitutive/incremental_linear_elastic_interface_law.cpp

```diff
@@ -0,0 +1,131 @@
+        mPreviousTraction +
+        prod(MakeConstitutiveMatrix(rValues.GetMaterialProperties()[INTERFACE_NORMAL_STIFFNESS],
+                                    rValues.GetMaterialProperties()[INTERFACE_SHEAR_STIFFNESS]),
+             rValues.GetStrainVector() - mPreviousRelativeDisplacement);
+}
+
+bool GeoIncrementalLinearElasticInterfaceLaw::RequiresInitializeMaterialResponse() { return false; }
+
+void GeoIncrementalLinearElasticInterfaceLaw::FinalizeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    mPreviousRelativeDisplacement = rValues.GetStrainVector();
+    mPreviousTraction             = rValues.GetStressVector();
+}
+
+int GeoIncrementalLinearElasticInterfaceLaw::Check(const Properties& rMaterialProperties,
+                                                   const ConstitutiveLaw::GeometryType& rElementGeometry,
+                                                   const ProcessInfo& rCurrentProcessInfo) const
+{
+    const auto result = BaseType::Check(rMaterialProperties, rElementGeometry, rCurrentProcessInfo);
+
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(INTERFACE_NORMAL_STIFFNESS))
+        << "No interface normal stiffness defined" << std::endl;
+
+    KRATOS_ERROR_IF_NOT(rMaterialProperties[INTERFACE_NORMAL_STIFFNESS] > 0.0)
+        << "Interface normal stiffness must be positive, but got "
+        << rMaterialProperties[INTERFACE_NORMAL_STIFFNESS] << std::endl;
+
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(INTERFACE_SHEAR_STIFFNESS))
+        << "No interface shear stiffness defined" << std::endl;
+
+    KRATOS_ERROR_IF_NOT(rMaterialProperties[INTERFACE_SHEAR_STIFFNESS] > 0.0)
+        << "Interface shear stiffness must be positive, but got "
+        << rMaterialProperties[INTERFACE_SHEAR_STIFFNESS] << std::endl;
+
+    KRATOS_ERROR_IF_NOT(dynamic_cast<const LineInterfaceGeometry*>(&rElementGeometry))
+        << "Expected a line interface geometry, but got " << rElementGeometry.Info() << std::endl;
+
+    return result;
+}
+
+void GeoIncrementalLinearElasticInterfaceLaw::save(Serializer& rSerializer) const
+{
+    KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, BaseType)
+    rSerializer.save("PreviousRelativeDisplacement", mPreviousRelativeDisplacement);
+    rSerializer.save("PreviousTraction", mPreviousTraction);
+}
+
+void GeoIncrementalLinearElasticInterfaceLaw::load(Serializer& rSerializer)
+{
+    KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, BaseType)
+    rSerializer.load("PreviousRelativeDisplacement", mPreviousRelativeDisplacement);
+    rSerializer.load("PreviousTraction", mPreviousTraction);
+}
+
+Matrix GeoIncrementalLinearElasticInterfaceLaw::MakeConstitutiveMatrix(double NormalStiffness,
+                                                                       double ShearStiffness) const
+{
+    auto result  = Matrix{ZeroMatrix{GetStrainSize(), GetStrainSize()}};
+    result(0, 0) = NormalStiffness;
+    result(1, 1) = ShearStiffness;
+    return result;
+}
```

Comment:
This is more of a personal preference, but I'd put this function close to where it's called (i.e. directly below CalculateMaterialResponseCauchy)

Reply:
The definitions of the member functions in the implementation file have the same ordering as the declarations in the header file. So first the `public` member functions, then the `private` member functions. The public members are all `override`s of the interface defined by class `ConstitutiveLaw`, and they are in the same order as in the header file of `ConstitutiveLaw`. From this point of view, I'm a bit reluctant to rearrange their order. Is that acceptable to you, or would you still insist on having the definition of `MakeConstitutiveMatrix` close to `CalculateMaterialResponseCauchy`?

Reply:
I get that reasoning as well. Since I think it improves readability, I still prefer moving it, but this is your PR and I'm also completely fine with choosing for consistency with the base class and header file.